CMakeLists.txt

```diff
@@ -70,6 +70,11 @@ IF(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
   ADD_COMPILE_OPTIONS("-Wno-psabi")
 ENDIF()
 
+# Xcode does not support COMPILE_FLAGS according to https://cmake.org/cmake/help/latest/prop_sf/COMPILE_FLAGS.html
```

Comment:
This is not what the linked document says: "However, Xcode does not support per-config per-source settings, so expressions that depend on the build configuration are not allowed with that generator." So, it is a combination of per-source-file and per-config settings that is not supported.

Reply:
I got it, It actually means it doesn't support Generator expression collaborating with Compile_flag, it will lead  to per-config per-source settings

Reply:
Fixed